docs/reference/troubleshooting/troubleshooting-unbalanced-cluster.asciidoc

```diff
@@ -0,0 +1,70 @@
+[[troubleshooting-unbalanced-cluster]]
+== Troubleshooting an unbalanced cluster
+
+Since 8.6 Elasticsearch is balancing shards across data tier to achieve a good compromise between
+* shard count
+* disk usage
+* ingest load forecast for write indices
+
+Please note, elasticsearch does not take into account amount or complexity of search queries when rebalancing shards.
+
+There is no guarantee that individual components are going to be evenly spread across the nodes.
+This might appear as some of the nodes have less shards allocated or using less disk
+as long as they are assigned with the shards with greater ingest load forecast.
+
+You might get a detailed output of assigned workloads per node using <<cat-allocation,cat allocation command>>.
```

Comment:
I'd suggest including this API call in the docs, similar to the desired balance API ones, so that users can just click on it.

Also could we make any simple additions to `GET _cat/allocation` to make it more useful in this context? For instance, would number of shards on undesired nodes, broken down by node, be a useful thing to include?

Reply:
Possibly, I was attempting a similar change before, but it is a little verbose and require node stats changes as well, see https://github.com/elastic/elasticsearch/pull/97561/files#r1259473590

Reply:
I don't think we'd need to do that for the number-of-shards-to-be-moved stat, we could either just execute the `TransportGetDesiredBalanceAction` or else introduce another master-node action dedicated to this task.

Reply:
Let me see if I can add that in https://github.com/elastic/elasticsearch/pull/105894

TransportGetDesiredBalanceAction is possible, but I would prefer to have cat/node stats rather then internal api

Reply:
Sure, but cat nodes and cat allocation already call several transport actions, I meant we could add `TransportGetDesiredBalanceAction` under the hood there